File: playwright-nykaa/flows/SearchFlow.ts
import { Page } from "@playwright/test";
import { HomePage } from "../pages/HomePage"

export class SearchFlow{
    readonly homePage: HomePage;
    constructor(private readonly page: Page) {
        this.homePage = new HomePage(this.page);
    }

    search = async (query: string) => {
        await this.homePage.goto();
        await this.homePage.searchProduct(query);
    }
}

File: week4selenide/src/test/java/com/week4selenide/flow/SearchFlow.java
package com.week4selenide.flow;

import com.week4selenide.pages.*;

public class SearchFlow{

    private CatalogPage catalogPage;
    private ProductPage productPage;

    public SearchFlow() {
        catalogPage = new CatalogPage();
        productPage = new ProductPage();
    }

    public CatalogPage openCatalogPage(){
        catalogPage.openPage();
        return catalogPage;
    }

    public CatalogPage Search(String query){
        catalogPage.searchProduct(query).verifyResult(query);
        return catalogPage;
    }

    public ProductPage clickOnFirstProduct(){
       productPage = catalogPage.openFirstProduct();
        return productPage;
    }

    public void verifyAvalibilityBadgeisVisible(){
        productPage.abilityBadgeVisible();
    }
}
